### PR TITLE
Core/197/revert bigchain deletes

### LINKED ISSUE #197: Prevent deletes on the bigchain table
Deletes are not permitted in the `bigchain` table. We need a changefeed to listen to delete in the `bigchain` table and revert them

### PR DESCRIPTION
**Pull request by rhsimplex**
Resolves #197, ready for review.

### CODE REVIEW

File: bigchaindb/core.py [hunk 1]
```diff
@@ -339,8 +343,17 @@ def validate_block(self, block):
...
-
-        # First: Run the plugin block validation logic
+        # First, make sure this node hasn't already voted on this block
+        if block['votes']:
+            for vote in block['votes']:
+                if vote['node_pubkey'] == self.me:
+                    if util.verify_vote_signature(block, vote):
+                        return block
+                    else:
+                        raise ImproperVoteError('Block {block_id} already has in incorrectly signed vote '
```
Comment (ttmc) [inline bigchaindb/core.py]: I believe you meant to write "...has an incorrectly signed vote"

File: bigchaindb/core.py [hunk 2]
```diff
@@ -449,6 +462,16 @@ def vote(self, block, previous_block_id, decision, invalid_reason=None):
...
         """Write the vote to the database."""
 
+        # First, make sure this block doesn't contain a vote from this node
+        if block['votes']:
+            for prev_vote in block['votes']:
+                if prev_vote['node_pubkey'] == self.me:
+                    if util.verify_vote_signature(block, prev_vote):
+                        return None
+                    else:
+                        raise ImproperVoteError('Block {block_id} already has in incorrectly signed vote '
```
Comment (ttmc) [inline bigchaindb/core.py]: Same comment as on line 353

**Comment by r-marques** [comment]
:+1:

**Comment by ttmc** [comment]
Note: Even though this branch was merged, the `BlockDeleteRevert` class is now gone. The wire protocol firewall (#594) will prevent deletes in the `bigchain` table.

**Pull request merged**